### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by gregoflash05**

### CODE REVIEW

File: auth/google.go [hunk 1]
```diff
@@ -18,10 +18,8 @@ import (
 	"zuri.chat/zccore/utils"
 )
 
-// var ClientId string = "107847729811-5gpmlkee2hm6urdlvm90l47nke6cg0og.apps.googleusercontent.com" // localhost:8080
-// var ClientSecret string = "gE5Si1htrsoPKOIUIeDaNZIn"                                             // localhost:8080
-var ClientId string = "107847729811-bc61afgl2s8tcqqae02dn67l27kue16o.apps.googleusercontent.com" // api.zuri.chat
-var ClientSecret string = "Zq972KMNCGA_pBkeHEZafpm2"                                             // api.zuri.chat
+var ClientId string = "107847729811-1i98f99fhamo3rg9930430tsua2he1ig.apps.googleusercontent.com" // api.zuri.chat
+var ClientSecret string = "W1sRuhVfIJzdJgO_Ryyj-pkQ"                                             // api.zuri.chat
```
Comment (samsoft00) [inline auth/google.go:22]: Why not move this to Config?

File: auth/models.go [hunk 2]
```diff
@@ -184,6 +190,16 @@ func IsAuthorized(user_id string, orgId string, role string, w http.ResponseWrit
 	var user user.User
 	mapstructure.Decode(userDoc, &user)
 
+	if role == "zuri_admin" {
+		if user.Role == role {
+			return true
+		} else {
+			utils.GetError(errors.New("Access Denied"), http.StatusUnauthorized, w)
+			return false
+		}
```
Comment (samsoft00) [inline auth/models.go:199]: User role suppose be a array, and also declare the list of roles as a constant in models

File: main.go [hunk 3]
```diff
@@ -61,13 +61,13 @@ func Router(Server *socketio.Server) *mux.Router {
...
 	// r.HandleFunc("/auth/template", auth.HtmlTemplate)
-	r.HandleFunc("/auth/{provider}/callback", auth.CompleteGoogleAuth).Methods("GET")
-	r.HandleFunc("/auth/{provider}", auth.BeginGoogleAuth)
 	// r.HandleFunc("/auth/test", auth.AuthTest).Methods(http.MethodPost)
 	r.HandleFunc("/auth/logout", auth.LogOutUser).Methods(http.MethodPost)
 	r.HandleFunc("/auth/logout/othersessions", auth.LogOutOtherSessions).Methods(http.MethodPost)
 	r.HandleFunc("/auth/verify-token", auth.IsAuthenticated(auth.VerifyTokenHandler)).Methods(http.MethodGet, http.MethodPost)
 	r.HandleFunc("/auth/confirm-password", auth.IsAuthenticated(auth.ConfirmUserPassword)).Methods(http.MethodPost)
+	r.HandleFunc("/auth/{provider}/callback", auth.CompleteGoogleAuth).Methods("GET")
+	r.HandleFunc("/auth/{provider}", auth.BeginGoogleAuth)
```
Comment (samsoft00) [inline main.go:70]: I think BeginGoogleAuth should be above the callback.

File: user/models.go [hunk 4]
```diff
@@ -64,6 +64,7 @@ type User struct {
 	Phone             string                 `bson:"phone" validate:"required" json:"phone"`
 	Settings          *UserSettings          `bson:"settings" json:"settings"`
 	Timezone          string                 `bson:"time_zone" json:"time_zone"`
+	Role              string                 `bson:"role" json:"role"`
```
Comment (samsoft00) [inline user/models.go:67]: Let's convert it to array/dictionary, and declare the array of roles in auth model

File: auth/google.go (hunk 1 above)
Comment (gregoflash05) [reply auth/google.go:22]: I am still testing this, I have changed the values and will keep changing it, leave it alone for now

File: auth/models.go (hunk 2 above)
Comment (gregoflash05) [reply auth/models.go:199]: There is no need for that at all, a user has just one role according to the authorisation implementation

File: main.go (hunk 3 above)
Comment (gregoflash05) [reply main.go:70]: I don't think the position of a function matters in Golang

File: user/models.go (hunk 4 above)
Comment (gregoflash05) [reply user/models.go:67]: Like I said earlier and have told Jb, their is no need to assign multipe roles to a user

File: auth/models.go (hunk 2 above)
Comment (samsoft00) [reply auth/models.go:199]: I think it's better to add it to config file or as a constant in auth/model, so anyone can refresh it as a single source of truth.

Comment (samsoft00) [reply auth/models.go:199]: I think it's better to add it to config file or as a constant in auth/model, so anyone can refrence it as a single source of truth.

**Pull request merged**